msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -2224,6 +2236,220 @@ public String processActionToggleDisplayExtendedDescription()
+            targetCalendar.commitEvent(end);
+          } else {  //if there is a Calendar record for the closing, but no close date, we interpret it as a removal.
+            targetCalendar.removeEvent(end);
+            calendarEndId = null;
+          }
+        } else if (calendarEndId == null){  //when there is not already a Calendar record for this closing date.
+          if (closeDate != null){ //don't put anything in the Calendar unless there is a Close date actually set.
+            end = targetCalendar.addEvent();
+            end.setDisplayName(closingTitle);
+            end.setDescription(getResourceBundleString("sendOpenCloseToCalendar.description.assembly",new Object[]{closingTitle,closeDate.toString()}));
+            end.setType(getResourceBundleString("sendOpenCloseToCalendar.type"));
+            end.setGroupAccess(allowedGroups, false);
+            end.setRange(this.timeService.newTimeRange(closeDate.getTime(), 0));
+            targetCalendar.commitEvent(end);
+          }
+        }
+        if (begin != null){ //put the Calendar entry for Open in the Forum data.
+          if (forumItem instanceof DiscussionTopic) {
+            ((DiscussionTopic) forumItem).setCalendarBeginId(begin.getId());
+          } else if (forumItem instanceof DiscussionForum) {
+            ((DiscussionForum) forumItem).setCalendarBeginId(begin.getId());
+          }
+        } else {
+          if (forumItem instanceof DiscussionTopic) {
+            ((DiscussionTopic) forumItem).setCalendarBeginId(null);
+          } else if (forumItem instanceof DiscussionForum) {
+            ((DiscussionForum) forumItem).setCalendarBeginId(null);
+          }
+        }
+        if (end != null) {  //put the Calendar entry for End in the forum data.
+          if (forumItem instanceof DiscussionTopic) {
+            ((DiscussionTopic) forumItem).setCalendarEndId(end.getId());
+          } else if (forumItem instanceof DiscussionForum) {
+            ((DiscussionForum) forumItem).setCalendarEndId(end.getId());
+          }
+        } else {
+          if (forumItem instanceof DiscussionTopic) {
+            ((DiscussionTopic) forumItem).setCalendarEndId(null);
+          } else if (forumItem instanceof DiscussionForum) {
+            ((DiscussionForum) forumItem).setCalendarEndId(null);
+          }
+        }
+      } else {  // when Send To Calendar is not checked
+        if (calendarBeginId != null){  //don't mess with the Calendar unless there have previously been Calendar events created for this topic
+          try {
+            begin = targetCalendar.getEditEvent(calendarBeginId, CalendarService.EVENT_MODIFY_CALENDAR);
+          } catch (IdUnusedException e) { //if we couldn't get Begin from the calendar, it likely means that the event was deleted from Calendar without consulting Forums, so Forums has an ID saved for a Calendar event that won't exist. We can just create it.
+            begin = targetCalendar.addEvent();
```

Comment:
Why bother adding this here? Why not just remove the event in the try block? It will never be reached if there is no event found.